app/models/concerns/nameable.rb

```diff
@@ -3,7 +3,7 @@
 module Nameable
   # class methods for Exercise/Pain/BodyPart models
   def by_name
-    order('lower(name) ASC')
+    order(Arel.sql('lower(name) ASC'))
   end
```

Comment:
Hi @Hunk13! And thanks for doing this. 

I'm not sure the `Arel.sql` will work for us. The error says "This method should not be called with user-provided values, such as request parameters or model attributes."  The `name` field is a user-provided value of a model attribute, so I think the `Arel.sql` fix is the equivalent to wrapping unsafe html in `html_safe`.

I'd like to figure out a way to sanitize those `name` inputs so that we can safely query against them.

Reply:
Hi @Hunk13,
Now that I've spent several hours researching my concern, I have a better understanding of what is happening. The `Arel` warning comes down to this: Rails is okay with ordering by the hard-coded column name `name`, as in: `order('name ASC')`, but it does not recognize `lower(name)` as the same thing as a plain, hardcoded `name`. This is why it is throwing the error. The `name` attribute is safe and the `lower(name)` is safe to pass into the `order` method. 
https://github.com/rails/rails/issues/32995 

I think the best course of action is go with your original submission, in restoring line 6 above to:
```
order(Arel.sql('lower(name) ASC'))
```
and removing the `sanitize` methods from the form inputs. 

I've learned a lot about security along the way. I hope you have too.

Reply:
@lortza the issue of application security is always interesting for me and I take it seriously. I think the solution to this issue was useful for us.